doc/reference/reactivity.md

```diff
@@ -2,58 +2,101 @@
 
-To solve this issue, Owl provides two reactivity primitives:
-
-- `reactive`, which returns a proxy to its first argument, and tracks all read/update
-  operation going through it,
-- `useState`: a hook, that internally uses `reactive`, and is linked to its
-  owner component: any read operation will be tracked (key by key), and any
-  updates to these tracked values will cause the component to be rerendered.
-
-Most of the time, the `useState` hook is the best solution.
-
-Since version 2.0, Owl applies the fine grained reactivity at the component
-level: reactive objects received as props are automatically subscribed to by the
-component, so Owl can track which part of these props are consumed by each
-component, and is therefore able to only rerender the impacted components.
+To this end, Owl provides a proxy-based reactivity system, based on the `reactive` primitive.
+The `reactive` function takes an object as a first argument, and an optional callback as its second argument,
+it will return a proxy of the object. This proxy will track what properties are read through the proxy,
+and call the provided callback whenever one of these properties is changed through any reactive version
+of the same object. It does so in depth, by returning reactive versions of the subobjects when they are read.
 
 ## `useState`
 
-Let us start by an example of how `useState` could be used:
+While the `reactive` primitive is very powerful, its usage in components follow a very standard pattern:
+components want to be rerendered when part of the state which they depend on for rendering changes. To
+this end, owl provides a standard hook: `useState`. To put it simply, this hook simply calls reactive
+with the provided object, and the current component's render function as its callback. This will cause
+it to rerender whenever any part of the state object that has been read by this component is modified.
+
+Here is a simple example of how `useState` can be used:
 
 ```js
 class Counter extends Component {
   static template = xml`
-    <div t-on-click="increment">
+    <div t-on-click="() => this.state.value++">
       <t t-esc="state.value"/>
     </div>`;
 
   setup() {
     this.state = useState({ value: 0 });
   }
+}
+```
+
+This component reads `state.value` when it renders, subscribing it to changes to that key. Whenever
+the value changes, Owl will update the component. Note that there is nothing special about the
+`state` property, you can name your state variables whatever you want, and you can have multiple of
+them on the same component if it makes sense to do so. This also allows `useState` to be used in custom
+hooks that may require state that is specific to that hook.
 
-  increment() {
-    this.state.value++;
+### Reactive props
+
+Since version 2.0, Owl renders are no longer "deep" by default: a component is only rerenderd by its
+parent if its props have changed (using a simple equality test). What if the contents of a props have
```

Comment:
seems weird to have "content" in plural form. not sure it is wrong, but is seems incorrect